### DISCUSSION TITLE
Feature to OpenROAD that lets us drag views with the mouse?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#3812] [General]
It would make moving around the design a lot easier.

Is there a way to do this already?

The arrow keys is a bit clunky compared to dragging the view with the mouse.

**Comment by stefanottili** [comment]
OpenRoad will hopefully take hint's from KLayout's layout view controls instead of inventing yet another scheme.

zoom in/out at the cursor position (mouse wheel zoom in/out)
make the cursor key "shift width" adjustable (40% seem like a better value then the defaults)
have "gestures" for fit, zoom in, zoom out (move mouse right-up, down, left-up), so no "mouse drag viewpoint".

There doesn't seem to be a consensus for zoom in/out, fit, level down/up keys.
I override the klayout defaults and use z/Z, f, l/L, curious what other people use.

**Comment by stefanottili** [comment]
I stand corrected, there is dragging the view in Klayout, on a Mac is "right mouse + shift"

**Comment by maliberty** [comment]
There is no method currently.  It could be an enhancement request as could gestures.

**Reply by oharboe** [reply]
Could you write a "help wanted" "good first issue" feature request?

It sounds like you have the language to describe these, what I assume are, standard features for EDA tools...

**Comment by oharboe** [comment]
Added feature request https://github.com/The-OpenROAD-Project/OpenROAD/issues/3840
